script.js

```diff
@@ -311,161 +322,200 @@ function paintScreen() {
+    var newItemElement = document.createElement('img');
+    newItemElement.id = items.length;
+    newItemElement.src = itemSrcs[itemType];
+    newItemElement.alt = `item${items.length}`;
+    newItemElement.style.position = "absolute";
+    newItemElement.style.width = 40 + 'px';
+    newItemElement.style.height = 40 + 'px';
+    items.push({ element: newItemElement, x: 0, y: 70, onBelt: true, inPackageId: null, type: itemType });
+
+
 }
-// function pickUpBins(){ //keyP
 
-// }
-//document.addEventListener('keypress');
+
 
 var viewportWidth = window.innerWidth;
 var viewportHeight = window.innerHeight;
 var playKeys = {}; // will store  key-true/false pairs 
 playKeys["Space"] = false;
+playKeys["KeyP"] = false;
 
 function keyBeingPressed(event) {
-    if (event.code != "Space") {
+    if (event.code != "Space" && event.code != "KeyP") {
         playKeys[event.code] = true;
         
     }
 }
 
 
 function keyNotBeingPressed(event) {
-    if (event.code != "Space") {
+    if (event.code != "Space" && event.code != "KeyP") {
         playKeys[event.code] = false;
     }
 }
 
-// function keyWasPressed(event){
-//     playKeys[event.code] = !playKeys[event.code];
-//     console.log(playKeys[event.code]);
-// }
+
 
 document.addEventListener('keydown', keyBeingPressed);
 document.addEventListener('keyup', keyNotBeingPressed);
 
 function movePlayer() { //edited in pickUpStuff
+    var changeInX = 0;
+    var changeInY = 0;
     if (playKeys['KeyW'] && playerY - 5 > 0) {
-        playerY -= 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.y -= 10;
-        }
+        changeInY -=10;
     }
     if (playKeys['KeyS'] && playerY + player.offsetHeight + 5 < viewportHeight) {
-        playerY += 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.y += 10;
-        }
+        changeInY +=10;
     }
 
     if (playKeys['KeyA'] && playerX - 5 > 0) {
-        playerX -= 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.x -= 10;
-        }
+        changeInX -=10;
     }
     if (playKeys['KeyD'] && playerX + player.offsetWidth + 5 < viewportWidth) {
-        playerX += 10;
-        if (playerIsCarrying != null) {
-            playerIsCarrying.x += 10;
-        }
+        changeInX +=10;
     }
+    playerY += changeInY;
+    playerX += changeInX;
     for (let i in items) {
-        if (items[i] == playerIsCarrying) {
+        if (items[i] == playerIsCarrying && playerIsCarrying != null) {
+            playerIsCarrying.x += changeInX;
+            playerIsCarrying.y += changeInY;
             items[i].x = playerIsCarrying.x;
             items[i].y = playerIsCarrying.y;
             items[i].element.style.left = items[i].x + 'px';
-            items[i].element.style.top = items[i].y; + 'px';
+            items[i].element.style.top = items[i].y + 'px';
         }
     }
+    for (let c in containers){
+        if (containers[c] == playerIsCarrying && playerIsCarrying != null) {
```

Comment:
Small thing, but you could check to see if `playerIsCarrying != null` before the entire for loop, since if that statement isn't true, you don't need to loop through the containers at all.